src/thread_task.rs

```diff
@@ -478,21 +499,22 @@ pub(crate) unsafe fn rav1d_task_create_tile_sbrow(
     }
     (*prev_t).next = 0 as *mut Rav1dTask;
     (*f).task_thread.done[(pass & 1) as usize].store(0, Ordering::SeqCst);
-    pthread_mutex_lock(&mut (*f).task_thread.pending_tasks.lock);
-    if !(((*f).task_thread.pending_tasks.head).is_null() || pass == 2) {
-        unreachable!();
-    }
-    if ((*f).task_thread.pending_tasks.head).is_null() {
-        (*f).task_thread.pending_tasks.head = &mut *tasks.offset(0) as *mut Rav1dTask;
-    } else {
-        (*(*f).task_thread.pending_tasks.tail).next = &mut *tasks.offset(0) as *mut Rav1dTask;
+    {
```

Comment:
Why do we need this block when the function ends right after the end of the block?

Reply:
Good point, we don't.